contracts/v2/PolygonZkEVMBridgeV2.sol

```diff
@@ -726,6 +735,34 @@ contract PolygonZkEVMBridgeV2 is
             ];
     }
 
+    /**
+     * @notice Set the address of a wrapper using the token information if already exist
+     * @dev This function is used to allow any existing token to be mapped with
+     *      origin token. Wrapper contract should handle mint/burn of the existing token.
+     * @param originNetwork Origin network
+     * @param originTokenAddress Origin token address, 0 address is reserved for ether
+     * @param wrappedTokenAddress Arbitary contract that implements TokenWrapped interface
+     */
+    function setCustomTokenMapping(
+        uint32 originNetwork,
+        address originTokenAddress,
+        address wrappedTokenAddress,
+        address existingTokenAddress
+    ) external onlyRollupManager {
```

Comment:
We should be using a role just for this. On the L2s there is no rollupManaer deployed.

I think I missed this in the original spec -- but perhaps it's something like a globalRootManager address or something